so3/devices/mem.c

```diff
@@ -0,0 +1,41 @@
+#include <vfs.h>
+#include <common.h>
+#include <process.h>
+#include <asm/io.h>
+#include <asm/mmu.h>
+#include <device/driver.h>
+
+#define DEV_CLASS_MEM "mem"
+
+static void *mem_mmap(int fd, addr_t virt_addr, uint32_t page_count, off_t offset)
+{
+	pcb_t *pcb;
+	bool no_cache;
+
+	pcb = current()->pcb;
+
+	/* Use cache only if the target offset is in RAM */
+	no_cache = (offset < mem_info.phys_base) &&
+	           ((mem_info.phys_base + mem_info.size) <= offset);
```

Comment:
```suggestion
	no_cache = ((offset + page_count * PAGE_SIZE) < mem_info.phys_base) &&
	           ((mem_info.phys_base + mem_info.size) <= offset);
```
Would this make sense ?